### ISSUE TITLE
Lost connection to device when used "onAuthorizer"

### ISSUE THREAD

**Issue by mustafaeida** [pusher/pusher-channels-flutter#107]
E/AndroidRuntime(32165): FATAL EXCEPTION: pusher-java-client eventQueue
E/AndroidRuntime(32165): Process: com.harajnet.app, PID: 32165
E/AndroidRuntime(32165): java.lang.NullPointerException: Attempt to invoke virtual method 'java.lang.String com.pusher.client.channel.impl.message.AuthResponse.getChannelData()' on a null object reference
E/AndroidRuntime(32165): 	at com.pusher.client.channel.impl.PrivateChannelImpl.authorize(PrivateChannelImpl.java:84)
E/AndroidRuntime(32165): 	at com.pusher.client.channel.impl.PrivateChannelImpl.toSubscribeMessage(PrivateChannelImpl.java:101)
E/AndroidRuntime(32165): 	at com.pusher.client.channel.impl.ChannelManager.lambda$sendOrQueueSubscribeMessage$0$com-pusher-client-channel-impl-ChannelManager(ChannelManager.java:134)
E/AndroidRuntime(32165): 	at com.pusher.client.channel.impl.ChannelManager$$ExternalSyntheticLambda0.run(Unknown Source:4)
E/AndroidRuntime(32165): 	at com.pusher.client.util.Factory.lambda$queueOnEventThread$0(Factory.java:140)
E/AndroidRuntime(32165): 	at com.pusher.client.util.Factory$$ExternalSyntheticLambda0.run(Unknown Source:2)
E/AndroidRuntime(32165): 	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1137)
E/AndroidRuntime(32165): 	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:637)
E/AndroidRuntime(32165): 	at java.lang.Thread.run(Thread.java:1012)
I/Process (32165): Sending signal. PID: 32165 SIG: 9

  ```

    dynamic onAuthorizer(String channelName, String socketId, dynamic options) async {
      await DioHelper.dio.post(
          'broadcasting/auth' ,
          data: {
            'socket_id'  : socketId,
            'channel_name'  : channelName,
          }).then((value){
  
        return {
          "auth": "16119b4b780e507a671d:9bd4d8d21178f5c77e06b8e6456385b6b03baeca821af9fc6af440cc19dace0c" // for testing 
        };
  
      }).catchError((error){
        return {
          "auth": "16119b4b780e507a671d:9bd4d8d21178f5c77e06b8e6456385b6b03baeca821af9fc6af440cc19dace0c" // for testing 
        };
      });
    }
```

**Comment by Zhamshid** [comment]
I also have such a problem, is there a solution?

**Comment by benjamin-tang-pusher** [comment]
Does this happen every time you subscribe to a private/presence channel? If so, its likely your auth endpoint is sending back an incorrect auth response which our library doesn't handle well. What is your auth response returning?

**Comment by benw-pusher** [comment]
This should have been resolved in 2.2.0 as a result of https://github.com/pusher/pusher-channels-flutter/pull/121
